コメントを付与したSQLが実行されない問題の修正

KazuyaUchida opened the pull request:
https://github.com/ray-di/Ray.QueryModule/pull/25 で追加されたオプションでSQL名を付与した場合と以下の様にSQLの先頭にコメントを挿入した場合に
`RowListInterface`ではSQLが実行されない問題を修正

## sample sql
```sql
/* todo_item_by_id_with_comment.sql */
SELECT
    *
FROM
    todo
WHERE
    id = :id 
```

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
	- Enhanced SQL query processing by removing comments and improving query validation.
	- Introduced new SQL files for selecting todo items by ID, including comments for clarity.

- **Tests**
	- Added multiple new test methods to validate the functionality of SQL queries with comments, ensuring expected results.
	- Refactored existing test methods for improved clarity and organization.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

KazuyaUchida committed:
コメントを付与したSQLが実行されない問題を修正

KazuyaUchida committed:
Feedback coderabbitai review

KazuyaUchida committed:
Feedback coderabbitai review / use phpunit function

koriym commented:
PHP7.4からのサポートなので一部関数がアウトだと思います。

KazuyaUchida committed:
rollback str_starts_with -> strpos

koriym commented:
@coderabbitai これまでのコメントを日本語でもお願いします。

koriym commented:
@coderabbitai いえ、あなたのコメントです。以下の２つを訳してください。

https://github.com/ray-di/Ray.QueryModule/pull/33#pullrequestreview-2510471483

57-57: Consider additional SQL comment handling improvements

The current implementation could be enhanced to:

Handle MySQL's # style comments
Better handle nested comments
Validate comment syntax correctness

64-80: Consider adding more comprehensive test cases.

While the current test methods cover basic comment scenarios, consider adding test cases for:

Single-line comments (--)
Multiple comments in different positions
Malformed or nested comments
Comments between SQL keywords

koriym commented:
@KazuyaUchida どうでしょうか？

KazuyaUchida commented:
@koriym 
coderabbitai の提案を試してみましたが、

> MySQLの # スタイルコメントへの対応

#スタイルコメントを挿入した場合は、そもそもSQLが実行できないです

```
PDOException : SQLSTATE[HY000]: General error: 1 unrecognized token: "#"
Ray.QueryModule/vendor/aura/sql/src/AbstractExtendedPdo.php:620
Ray.QueryModule/vendor/aura/sql/src/AbstractExtendedPdo.php:554
Ray.QueryModule/src/SqlQueryRowList.php:52
```

>  64-80: Consider adding more comprehensive test cases.

いくつか提案はされていますが、作成したテストで用意したSQLでカバーできています
今回の対応は先頭にコメントが挿入された場合に実行されてないケースの対応のため、SQL内のコメントがあった場合はスコープ外の認識です

>  1行コメント（--）

`testWithLineComment`でチェック済み
SQLとして以下の様な書き方をしている場合はコメントアウトされていると思われるので、SQLが実行される方が問題かと思われます
```sql
-- todo_item_by_id_with_line_comment.sql SELECT * FROM todo WHERE id = :id
```

> 異なる場所に複数のコメントが含まれる場合

一応`todo_item_by_id_with_comment.sql`で先頭以外のブロックコメントが含まれている場合のSQLでチェックしています


PHPStanでエラーになっているところも、今回の修正とは関係のないところでエラーになっているので、対応するか悩みどころです💦

koriym commented on tests/Fake/sql/todo_item_by_id_with_comment.sql:
```diff
@@ -0,0 +1,7 @@
+FROM
+    todo /* table */
+WHERE
+    id = :id  /* conditions */
```
```suggestion
    id = :id  /* conditions */

```

koriym commented on tests/Fake/sql/todo_item_by_id_with_line_comment.sql:
```diff
@@ -0,0 +1,2 @@
+-- todo_item_by_id_with_line_comment.sql
+SELECT * FROM todo WHERE id = :id -- comment
```
```suggestion
SELECT * FROM todo WHERE id = :id -- comment

```

koriym commented on tests/Fake/sql/todo_item_by_id_with_multiple_comment.sql:
```diff
@@ -0,0 +1 @@
+/* todo_item_by_id_with_multiple_comment.sql */ SELECT * FROM todo WHERE id = :id  -- conditions
```
```suggestion
/* todo_item_by_id_with_multiple_comment.sql */ SELECT * FROM todo WHERE id = :id  -- conditions

```

koriym reviewed (changes requested):
EOLの追加をsuggestしました。

KazuyaUchida committed:
Apply suggestions from code review

Co-authored-by: Akihito Koriyama <akihito.koriyama@gmail.com>

KazuyaUchida commented:
@koriym 
https://github.com/ray-di/Ray.QueryModule/pull/33/commits/eaf52fa1d47af23dbf4b718fe07890de1eaf07d6 で反映しました！

koriym committed:
Add type annotations to QueryInterceptor methods

Added precise type annotations for MethodInvocation and other parameters in QueryInterceptor. This improves code readability and aids in static analysis. No functional changes were made.

koriym committed:
Update src/QueryInterceptor.php

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

koriym committed:
fixup! Add type annotations to QueryInterceptor methods

koriym committed:
Refactor SQL cleanup by introducing reusable constants

Introduce `QUERY_CLEANUP_REGEX` and `TRIM_CHARACTERS_REGEX` constants to improve code readability and reduce redundancy. Replace hardcoded regex strings with these constants in SQL trimming and cleanup operations. This makes future maintenance and updates to these patterns simpler and more centralized.

koriym committed:
Refactor type annotations for query result handling.

Updated type annotations to ensure compatibility with diverse data structures and improve clarity. This change enhances maintainability and conforms to best practices for type declarations.

koriym commented:
> PHPStanでエラーになっているところも、今回の修正とは関係のないところでエラーになっているので、対応するか悩みどころです💦

修正しました。
